qemu/tests/block_hotplug.py

```diff
@@ -21,144 +22,179 @@ def run(test, params, env):
+                pause):
+            test.fail(
+                "Failed to unplug devices from guest, need to unplug: %d, "
+                "actual unplug: %d" % (extra_disks_num, len(unplug_disks)))
+        session.close()
+
+    def format_disk_win():
+        """ Format disk in windows. """
+        error_context.context("Format disk %s in windows." % disk, logging.info)
+        session = vm.wait_for_login(timeout=timeout)
+        if disk_index is None and disk_letter is None:
+            drive_letters.append(
+                utils_disk.configure_empty_windows_disk(
+                    session, disk, params['image_size_%s' % imgs[index]])[0])
+        elif disk_index and disk_letter:
+            # For run "block_resize" sub test, assign disk_index and disk_letter.
+            utils_misc.format_windows_disk(
+                session, disk_index[index - 1], disk_letter[index - 1])
+        session.close()
+
+    def _add_assign_letter():
+        """ Add assign option for umount_disk in disk_update_cmd. """
+        params['disk_update_cmd'] = re.sub(r'--op=umount_disk',
+                                           '--op=umount_disk --assign=%s' %
+                                           drive_letters[0],
+                                           params['disk_update_cmd'])
+
+    def disk_io_test():
+        """ Do disk io test on disk. """
+        error_context.context("Do disk io test on hotplug disk.", logging.info)
+        session = vm.wait_for_login(timeout=timeout)
+        if os_type == "linux":
+            test_cmd = disk_op_cmd % (disk, disk)
+        elif os_type == "windows":
+            if iteration == 0:
+                format_disk_win()
+                if params.get('disk_update_cmd'):
+                    _add_assign_letter()
+            drive_letter = drive_letters[index - 1]
+            test_cmd = disk_op_cmd % (drive_letter, drive_letter)
+            test_cmd = utils_misc.set_winutils_letter(session, test_cmd)
+        else:
+            test.cancel("Unsupported OS type '%s'" % os_type)
+        session.cmd(test_cmd, timeout=disk_op_timeout)
+        session.close()
+
+    imgs = params.get("images").split()
+    os_type = params["os_type"]
     #sometimes, ppc can't get new plugged disk in 5s, so time to 10s
     pause = float(params.get("virtio_block_pause", 10.0))
-    blk_num = int(params.get("blk_num", 1))
+    extra_disks_num = len(imgs) - 1
     repeat_times = int(params.get("repeat_times", 3))
     timeout = int(params.get("login_timeout", 360))
+    disk_op_cmd = params.get("disk_op_cmd")
     disk_op_timeout = int(params.get("disk_op_timeout", 360))
-    get_disk_cmd = params.get("get_disk_cmd")
-    context_msg = "Running sub test '%s' %s"
     disk_index = params.objects("disk_index")
     disk_letter = params.objects("disk_letter")
+    blk_num = params.get("blk_num")
+    data_disks_num = int(blk_num) + 1 if blk_num else len(imgs)
```

Comment:
`imgs[0]` is always skipped, why not directly retrieve data disks from `imgs` instead of calculating the indexes.

Reply:
like:
```
data_disks = imgs[1:blk_num+1] if blk_num else imgs[1:]
```

Reply:
Or better to use `operator.itemgetter` to directly operates on `imgs`.